### PR TITLE
Fixed imports and documentation in `sympy.concrete` and `sympy.tensor`

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Removed unused imports, sorted imports, added and corrected sectioning in documentation.

Probably fixed an undetected bug in `sympy.concrete.delta.py`. Couldn't figure out how to trigger it, so didn't add a test, but once the line with `Sum` was called it would not have worked.

#### Other comments
Good way to browse the code.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/concrete/delta.py [hunk 1]
```diff
@@ -317,6 +317,7 @@ def deltasummation(f, limit, no_piecewise=False):
     if len(solns) == 0:
         return S.Zero
     elif len(solns) != 1:
+        from sympy.concrete.summations import Sum
         return Sum(f, limit)
```
Comment (sidhantnagpal) [inline sympy/concrete/delta.py]: Can you add a test for this change? (clearly, it wasn't tested before)

Comment (oscargus) [reply sympy/concrete/delta.py]: I've spent some time and cannot figure out a case where there are multiple solutions to the solve. Neither the polynomials or trigonometric functions I tried gave anything.

Comment (asmeurer) [reply sympy/concrete/delta.py]: It looks like `_extract_delta` explicitly excludes deltas from the delta term if they can have more than one solution (see `_is_simple_delta`). So I think this code path would only ever be hit if there is a bug, or if the above code changed.

Comment (asmeurer) [reply sympy/concrete/delta.py]: In other words I think this can be merged as is.

**Pull request merged**